lib/runner.js

```diff
@@ -202,10 +203,13 @@ Runner.prototype.fail = function(test, err) {
   ++this.failures;
   test.state = 'failed';
 
-  if (!(err instanceof Error)) {
+  var isError = err instanceof Error;
+  var isErrorLike = err && isString(err.message) && isString(err.name);
```

Comment:
@boneskull Sorry, I fail to see how it will throw an exception when `err` is `null` :( This test passes for me:
```js
it('should emit a helpful message when failed with null', function(done){
  var test = {}, err = null;
  runner.on('fail', function(test,err){
    err.message.should.equal('the null [null] was thrown, throw an Error :)');
    done();
  });
  runner.fail(test, err);
})
```
The message could definitely be more helpful, but no crash.